tests/test_mgmtvrf.py

```diff
@@ -9,6 +9,10 @@
 
 logger = logging.getLogger(__name__)
 
+# fixtures
+@pytest.fixture(scope="module")
+def localhost(testbed_devices):
+    return testbed_devices['localhost']
 
 @pytest.fixture(scope='module',autouse=True)
 def setup_mvrf(duthost, testbed_devices, testbed, localhost):
```

Comment:
@irene-pan1202 I think there is no need to define the localhost fixture again in the script. You can use the localhost object already in the `testbed_devices` fixture supplied to setup_mvrf(). For example, in setup_mvrf():
```
    localhost = testbed_devices["localhost"]
```
BTW, the `localhost` supplied to setup_mvrf() is redundant and can be removed.

Reply:
As you say, other test items that require localhost need to use the localhost object in the testbed_devices fixture individually?
But at #1432, localhost = testbed_devices ['localhost'] has been replaced by localhost in test_snmp_fact test case. Which way is better?
Just like test_vrf.py, because many test items need to use localhost, so define the localhost fixture?

Reply:
Currently there are two ways to get access to the localhost object. The first one is to get it from testbed_devices fixture. The second one is defined in https://github.com/Azure/sonic-mgmt/blob/master/tests/common/plugins/ansible_fixtures.py
IMO, both of these definitions need improvement. 
There was discussion about improving testbed_devices fixture here: https://github.com/Azure/sonic-mgmt/pull/1475#discussion_r396295577
The localhost fixture defined in ansible_fixtures.py should be moved to tests/conftest.py.

But anyway, the localhost object is already defined (already defined twice), there is no need to define it again in the test script.

>Localhost is not defined before using when setup Management vrf configs before the start of testsuite, it will cause 'Host Unreachable' error

I am not quite sure about your statement of the root cause. The localhost is defined indeed. Did you get a chance to test your fix? According to https://github.com/Azure/sonic-mgmt/issues/1615#issuecomment-625297443, the setup mgmt vrf failed because the console hangs after execution of command `sudo config vrf add mgmt`.